usaspending_api/search/filters/postgres/tas.py

```diff
@@ -0,0 +1,112 @@
+from usaspending_api.common.exceptions import InvalidParameterException
+from usaspending_api.accounts.models import TreasuryAppropriationAccount, FederalAccount
+from django.db.models import Q
+from usaspending_api.search.filters.postgres.HierarchicalFilter import HierarchicalFilter, Node
+from usaspending_api.accounts.helpers import TAS_COMPONENT_TO_FIELD_MAPPING
+
+
+class TasCodes(HierarchicalFilter):
+    underscore_name = "tas_codes"
+
+    @classmethod
+    def build_tas_codes_filter(cls, queryset, filter_values):
+        if isinstance(filter_values, list):
+            # This is a legacy usage, and will be dealt with by the other filter
+            return TreasuryAccounts.build_tas_codes_filter(queryset, filter_values)
+        elif isinstance(filter_values, dict):
+            require = filter_values.get("require") or []
+            exclude = filter_values.get("exclude") or []
+        else:
+            raise InvalidParameterException(f"tas_codes must be an array or object")
+
+        return Q(
+            treasury_account_identifiers__overlap=list(
+                cls._query_string(TreasuryAppropriationAccount.objects.all(), require, exclude).values_list(
+                    "treasury_account_identifier", flat=True
+                )
+            )
+        )
+
+    @staticmethod
+    def node(code, positive, positive_naics, negative_naics):
+        return TASNode(code, positive, positive_naics, negative_naics)
+
+
+def string_to_dictionary(string):
```

Comment:
Very similar to `string_to_dictionary()` in  usaspending_api/search/filters/elasticsearch/tas.py